waspc/cli/src/Wasp/Cli/Command/BuildStart.hs

```diff
@@ -0,0 +1,175 @@
+module Wasp.Cli.Command.BuildStart
+  ( buildStart,
+  )
+where
+
+import Control.Concurrent.Async (concurrently)
+import Control.Concurrent.Chan (newChan)
+import Control.Monad.Except (ExceptT (ExceptT), runExceptT)
+import Control.Monad.IO.Class (liftIO)
+import Data.Char (isAsciiLower, isAsciiUpper, isDigit, toLower)
+import Data.Function ((&))
+import StrongPath ((</>))
+import qualified StrongPath as SP
+import System.Process (proc)
+import System.Random (Random (randoms), RandomGen, newStdGen)
+import Wasp.AppSpec (AppSpec)
+import qualified Wasp.AppSpec.Valid as ASV
+import Wasp.Cli.Command (Command, require)
+import Wasp.Cli.Command.Compile (analyze)
+import Wasp.Cli.Command.Message (cliSendMessageC)
+import Wasp.Cli.Command.Require (BuildDirExists (BuildDirExists), InWaspProject (InWaspProject))
+import Wasp.Cli.Message (cliSendMessage)
+import Wasp.Generator.Common (ProjectRootDir)
+import Wasp.Generator.ServerGenerator.Common (defaultDevServerUrl)
+import Wasp.Generator.WebAppGenerator.Common (defaultClientPort, getDefaultDevClientUrl)
+import qualified Wasp.Generator.WebAppGenerator.Common as Common
+import qualified Wasp.Job as J
+import Wasp.Job.Except (JobExcept, toJobExcept)
+import qualified Wasp.Job.Except as JobExcept
+import Wasp.Job.IO (readJobMessagesAndPrintThemPrefixed)
+import Wasp.Job.Process (runNodeCommandAsJob, runNodeCommandAsJobWithExtraEnv, runProcessAsJob)
+import qualified Wasp.Message as Msg
+import Wasp.Project.Common (WaspProjectDir, buildDirInDotWaspDir, dotWaspDirInWaspProjectDir, makeAppUniqueId)
+import Wasp.Project.Env (dotEnvServer)
+
+buildStart :: Command ()
+buildStart = do
+  InWaspProject waspProjectDir <- require
+  appSpec <- analyze waspProjectDir
+  BuildDirExists <- require
+
+  -- TODO: Find a way to easily check we can connect to the DB.
+  -- Right now we just assume it is running and let Prisma fail if it is not. It
+  -- is not easy for us to do the same check as in other commands
+  -- (`DBConnecionEstablished <- require`), because that needs a built Prisma
+  -- schema, and currently we do that inside the Dockerfile, not in the
+  -- `.wasp/build` folder, like the `wasp start` command does.
+  -- It is not a big problem right now, because Prisma will fail shortly after
+  -- the server starts if the DB is not running anyway, and with a very clear
+  -- error message that we print.
```

Comment:
Related: https://github.com/wasp-lang/wasp/pull/2820#issuecomment-2980771384

Reply:
> idea: s/TODO/TODO(carlos) -> can be useful because people know whom to ask later if they need some clarification, and can't really hurt.

Uncle Bob would say this is sufficiently encoded in the VCS 😂 But I don't mind